subsys/bluetooth/mesh/proxy.c

```diff
@@ -1240,7 +1240,7 @@ s32_t bt_mesh_proxy_adv_start(void)
 	BT_DBG("");
 
 	if (gatt_svc == MESH_GATT_NONE) {
-		return K_FOREVER;
+		return K_TICKS_FOREVER;
```

Comment:
This doesn't look right either. Formally `K_TICKS_*` should indicate a number of ticks whereas this function is supposed to return a number of milliseconds. Shouldn't it be possible to just change `bt_mesh_proxy_adv_start()` to return a `k_timeout_t`?

Reply:
Yes, did that instead.